### PR TITLE
[WIP] Doctrine integration

### PR DESCRIPTION
**Pull request by WouterSioen**
I'm integrating Doctrine in Fork. I'll keep a checklist here containing my progress.

Every help is welcome. If you want to help but don't know to start where, feel free to poke me (here or on IRC), or take a look at the code in this PR.

## Modules

- [ ] Analytics module
- [ ] Blog module
- [x] Content Blocks module (done by @WouterSioen)
- [ ] Extensions module
- [ ] Faq module (@WouterSioen is working on this module)
- [ ] FormBuilder module
- [ ] Groups module (@MathiasDewelde is working on this module)
- [ ] Locale module
- [x] Location module (done by @WouterSioen)
- [ ] Mailmotor module
- [ ] Pages module
- [ ] Profiles module
- [ ] Search module
- [ ] Settings module
- [x] Tags module (done by @jeroendesloovere)
- [ ] Users module

## Core

- [ ] emails
- [ ] meta (Using [Sluggable](https://github.com/Atlantic18/DoctrineExtensions/blob/master/doc/sluggable.md) for URL creating?)
- [ ] modules
- [ ] modules settings
- [X] modules tags (done by @jeroendesloovere)
- [ ] timezones
- [ ] groups
- [ ] rights
- [ ] backend navigation
- [ ] hooks (queue and subscriptions) (Will this still be used when Symfony EventDispatcher is integrated?)

## Notes

- Use [Translatable](https://github.com/Atlantic18/DoctrineExtensions/blob/master/doc/translatable.md) for multi-language fields.
- Use [Sortable](https://github.com/Atlantic18/DoctrineExtensions/blob/master/doc/sortable.md) instead of **sequence**.
- Use [Tree](https://github.com/Atlantic18/DoctrineExtensions/blob/master/doc/tree.md) for multi-level categories.

### CODE REVIEW

File: src/Backend/Modules/Faq/Entity/Category.php [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+     * @var int
+     *
+     * @ORM\Column(type="integer")
+     */
+    private $extraId;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(type="string", length=255)
```
Comment (jeroendesloovere) [inline src/Backend/Modules/Faq/Entity/Category.php]: Shouldn't this be ```length=5```?

**Comment by jeroendesloovere** [comment]
@WouterSioen In BackendLocationModel, Doctrine is not yet integrated at function exists($id)

**Comment by MathiasDewelde** [comment]
I think the "exists" functions shouldn't be used anymore if the module is using Doctrine.

**Comment by jeroendesloovere** [comment]
@WouterSioen 
I just used the new class for Doctrine DataGrids: ``DataGridDoctrine();```.
Thats for basic Entities.
Is there already a class available for more advanced Entities with joins?

**Comment by WouterSioen** [comment]
Nope

**Comment by jeroendesloovere** [comment]
@WouterSioen 
I've been experimenting with [Doctrine Extension Translatable](https://github.com/Atlantic18/DoctrineExtensions/blob/master/doc/translatable.md#example-of-multiple-translations).
Perhaps this PR is a good time to integrate such a feature in all modules?

**Comment by jeroendesloovere** [comment]
As discussed with @WouterSioen, this PR will be changed a lot. Since we're opting to use [Embeddables](https://doctrine-orm.readthedocs.org/en/latest/tutorials/embeddables.html) to reach to the highest level of OOP.

**Comment by WouterSioen** [comment]
We'll introduce value objects and immutable entities to make sure our objects can't be invalid. No more entities with setters for every property. This is against good OOP design. (and allows you to create for example a blogpost without a title and a text).

**Comment by jeroendesloovere** [comment]
Note to @WouterSioen, if you want to use ValueObjects, you need to use Doctrine/ORM > v2.5.
Which has a downside: you can only use it when having PHP version 5.4 or greater.
So that is a bummer for users on older hostings...

**Comment by WouterSioen** [comment]
We've bumped the minimum php version to 5.4 in the latest release.

Op woensdag 15 juli 2015 heeft Jeroen Desloovere <notifications@github.com>
het volgende geschreven:

**Comment by jeroendesloovere** [comment]
[The "3.10 branch" composer.json](https://github.com/forkcms/forkcms/blob/3.10/composer.json) file still contains `>=5.3.2`.

**Comment by WouterSioen** [comment]
The master will be merged in 3.10 again later ;)

Op woensdag 15 juli 2015 heeft Jeroen Desloovere <notifications@github.com>
het volgende geschreven:

**Pull request closed without merging**